[Pytorch] gfx1153 - Pytorch build failures in all versions "error: use of undeclared identifier 'CK_BUFFER_RESOURCE_3RD_DWORD'"

### Run:
- https://github.com/ROCm/TheRock/actions/runs/20592874237
- https://github.com/ROCm/TheRock/actions/runs/20615762593

**Platform:** `Linux, Windows`
**Gfx Effected:** `gfx1153`
**Python Versions:** `3.11, 3.12, 3.13, 3.14`
**Pytorch Versions:** `2.7, 2.8, 2.9, nightly`

### Jobs:
- https://github.com/ROCm/TheRock/actions/runs/20592874237/job/59141347406
- https://github.com/ROCm/TheRock/actions/runs/20592874237/job/59141347465
- https://github.com/ROCm/TheRock/actions/runs/20592874237/job/59141347435
- https://github.com/ROCm/TheRock/actions/runs/20615762593/job/59208133548
- https://github.com/ROCm/TheRock/actions/runs/20615762593/job/59208133558

### Error:

```
here
2025-12-30T09:26:23.4326437Z   169 | #define __assert_fnuz_support(interp)                                                              \
2025-12-30T09:26:23.4327293Z       |         ^
2025-12-30T09:26:23.4328576Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/native/hip/bgemm_kernels/bgemm_kernel_bf16bf16bf16_128_16x64x64_16x16_1x2_8x16x1_8x16x1_1x16x1x8_4_Intrawave_v2.hip:3:
2025-12-30T09:26:23.4330479Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/native/hip/bgemm_kernels/bgemm_kernel_template.h:11:
2025-12-30T09:26:23.4332568Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/tensor_operation/gpu/device/impl/device_batched_gemm_multiple_d_xdl_cshuffle_v3.hpp:9:
2025-12-30T09:26:23.4334942Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/utility/common_header.hpp:39:
2025-12-30T09:26:23.4336951Z /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/utility/amd_buffer_addressing.hpp:32:48: error: use of undeclared identifier 'CK_BUFFER_RESOURCE_3RD_DWORD'
2025-12-30T09:26:23.4338314Z    32 |     wave_buffer_resource.config(Number<3>{}) = CK_BUFFER_RESOURCE_3RD_DWORD;
2025-12-30T09:26:23.4338923Z       |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-12-30T09:26:23.4340387Z /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/utility/amd_buffer_addressing.hpp:47:48: error: use of undeclared identifier 'CK_BUFFER_RESOURCE_3RD_DWORD'
2025-12-30T09:26:23.4342014Z    47 |     wave_buffer_resource.config(Number<3>{}) = CK_BUFFER_RESOURCE_3RD_DWORD;
2025-12-30T09:26:23.4342632Z       |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-12-30T09:26:23.4344024Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/native/hip/bgemm_kernels/bgemm_kernel_bf16bf16bf16_128_16x64x64_16x16_1x2_8x16x1_8x16x1_1x16x1x8_4_Intrawave_v2.hip:3:
2025-12-30T09:26:23.4346320Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/native/hip/bgemm_kernels/bgemm_kernel_template.h:11:
2025-12-30T09:26:23.4348230Z In file included from /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/tensor_operation/gpu/device/impl/device_batched_gemm_multiple_d_xdl_cshuffle_v3.hpp:15:
2025-12-30T09:26:23.4350881Z /__w/TheRock/TheRock/external-builds/pytorch/pytorch/aten/src/ATen/../../../third_party/composable_kernel/include/ck/tensor_operation/gpu/grid/gridwise_gemm_xdl_cshuffle_v3_multi_d.hpp:762:35: warning: division by zero is undefined [-Wdivision-by-zero]
2025-12-30T09:26:23.4352917Z   762 |                     ? KThreadRead / (kfold * K0PerThreadWrite / K0PerThreadRead)
```

--- a/.github/workflows/release_portable_linux_pytorch_wheels.yml
+++ b/.github/workflows/release_portable_linux_pytorch_wheels.yml
@@ -94,6 +94,8 @@ jobs:
         # redefine arch as inputs.x vars are not available in matrix context
         arch: ["${{ inputs.amdgpu_family }}"]
         exclude:
+        # gfx1153 support is only available in pytorch 2.10+
+        # see https://github.com/ROCm/TheRock/issues/2723
         - arch: "gfx1153"
           pytorch_git_ref: "release/2.8"
         - arch: "gfx1153"

--- a/.github/workflows/release_windows_pytorch_wheels.yml
+++ b/.github/workflows/release_windows_pytorch_wheels.yml
@@ -94,6 +94,8 @@ jobs:
         # redefine arch as inputs.x vars are not available in matrix context
         arch: ["${{ inputs.amdgpu_family }}"]
         exclude:
+        # gfx1153 support is only available in pytorch 2.10+
+        # see https://github.com/ROCm/TheRock/issues/2723
         - arch: "gfx1153"
           pytorch_git_ref: "release/2.9"
 

